### PR TITLE
EKS detector

### PR DESCRIPTION
**Pull request by GuyMoses**

### CODE REVIEW

File: src/lumigo_opentelemetry/__init__.py [hunk 1]
```diff
@@ -30,6 +30,11 @@ def _setup_logger(logger_name: str = "lumigo-opentelemetry") -> logging.Logger:
         logging.CRITICAL
     )
 
+    # Suppress spurious warnings when the application is not running on EKS
```
Comment (nirLumigo) [inline src/lumigo_opentelemetry/__init__.py:33]: Don't you need to add `if` to verify we're not inside an EKS?

File: src/test/unit/resources/eks_utils.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+MockEksResourceAttributes = {
+    ResourceAttributes.CLOUD_PROVIDER: CloudProviderValues.AWS.value,
+    ResourceAttributes.CLOUD_PLATFORM: CloudPlatformValues.AWS_EKS.value,
+    ResourceAttributes.K8S_CLUSTER_NAME: "mock-cluster-name",
+    ResourceAttributes.CONTAINER_ID: "a4d00c9dd675d67f866c786181419e1b44832d4696780152e61afd44a3e02856",
+}
+
+
+get_cluster_info = f"""{{
```
Comment (nirLumigo) [inline src/test/unit/resources/eks_utils.py:15]: Usually when importing a const it's upper case and a function is a lower case
```suggestion
GET_CLUSTER_INFO = f"""{{
```

File: src/test/unit/resources/eks_utils.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }}
+  }},
+  "data": {{
+    "cluster.name": "{MockEksResourceAttributes[ResourceAttributes.K8S_CLUSTER_NAME]}",
+    "logs.region": "us-west-2"
+  }}
+}}
+"""
+
+container_id_text = f"""14:name=systemd:/docker/{MockEksResourceAttributes[ResourceAttributes.CONTAINER_ID]}
```
Comment (nirLumigo) [inline src/test/unit/resources/eks_utils.py:36]:
```suggestion
CONTAINER_ID_TEXT = f"""14:name=systemd:/docker/{MockEksResourceAttributes[ResourceAttributes.CONTAINER_ID]}
```

File: src/test/unit/resources/test_detectors.py [hunk 4]
```diff
@@ -127,3 +129,35 @@ def test_get_resource_aws_ecs_resource_detector_not_ecs_container(caplog):
...
+    mock_is_eks,
+    mock_get_k8_cred_value,
+):
+    resource = get_resource({})
+
+    assert resource.attributes[ResourceAttributes.CLOUD_PROVIDER] == "aws"
+    assert resource.attributes[ResourceAttributes.CLOUD_PLATFORM] == "aws_eks"
+    assert isinstance(resource.attributes[ResourceAttributes.K8S_CLUSTER_NAME], str)
+    assert len(resource.attributes[ResourceAttributes.K8S_CLUSTER_NAME]) > 1
+    assert isinstance(resource.attributes[ResourceAttributes.CONTAINER_ID], str)
```
Comment (nirLumigo) [inline src/test/unit/resources/test_detectors.py:163]: Why not add
`assert len(resource.attributes[ResourceAttributes. CONTAINER_ID]) > 1`
as well?

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Nice! :)

**Review by nirLumigo — APPROVED** [review]

**Review by mmanciop — APPROVED** [review]

**Pull request merged**